### PR TITLE
#39 - Latest replacement using path uri

### PR DESCRIPTION
**Pull request by HDouss**
Resolving issue #39

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#41](https://github.com/artipie/go-adapter/issues/41) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/bea77ec1-2937-4648-8a01-acd336469b82); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/http/LatestSlice.java [hunk 1]
```diff
@@ -70,13 +69,28 @@ public Response response(
...
             )
         );
     }
 
+    /**
+     * Replaces the word latest if it is the last part of the URI path, by v. Then returns the path.
+     * @param line Received request line
+     * @return A URI path with replaced latest.
+     */
+    private static String toV(final String line) {
```
Comment (Vatavuk) [inline src/main/java/com/artipie/http/LatestSlice.java:84]: @HDouss it's not really clear from the method name what it does. How about `normalizedPath` or `pathWithVersion`?

Comment (HDouss) [reply src/main/java/com/artipie/http/LatestSlice.java:84]: @Vatavuk I agree, the name is bad, I'll try to find out a better one...

**Comment by HDouss** [comment]
@Vatavuk Thanks. Fixed!

**Comment by 0crat** [comment]
Job was finished in 45 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@dc9a75c2-99a9-40a1-bc27-ddaf8f08c10c, thread: pool-906-thread-1
java.lang.Thread[-2] java.lang.OutOfMemoryError: unable to create new native thread
```

[0.53.12](https://github.com/zerocracy/farm): Issue: artipie/go-adapter#41, Comment: 619908839

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@dc9a75c2-99a9-40a1-bc27-ddaf8f08c10c, thread: pool-906-thread-1
com.zerocracy.radars.github.ReSafe[81] java.io.IOException: java.lang.OutOfMemoryError: unable to create new native thread
java.lang.Thread[-2] java.lang.OutOfMemoryError: unable to create new native thread
```

[0.53.12](https://github.com/zerocracy/farm): Issue: artipie/go-adapter#41, Action: `created`

**Comment by Vatavuk** [comment]
@sereshqua please try again

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**